### PR TITLE
Fix energy calculations for QMs with squared terms

### PR DESCRIPTION
**Pull request by arcondello**
Closes https://github.com/dwavesystems/dimod/issues/1136

### CODE REVIEW

**Comment by alexzucca90** [comment]
do we need to modify this part too?
https://github.com/dwavesystems/dimod/blob/ffee615fe6b48a829d44e455fd9fe9561a1496ec/dimod/include/dimod/quadratic_model.h#L410

**Comment by arcondello** [comment]
Good catch, we should. Though that's not used from the Python level

File: tests/test_quadratic_model.py [hunk 1]
```diff
@@ -296,6 +296,11 @@ def test_bug982(self):
         self.assertAlmostEqual((i*j).energy({'i': 4294967296, 'j': 4294967296}),
                                1.8446744073709552e+19)
 
+    def test_bug1136(self):
+        # https://github.com/dwavesystems/dimod/issues/1136
+        i = dimod.Integer('i')
+        self.assertEqual((i**2).energy({'i': 1}), 1)
```
Comment (alexzucca90) [inline tests/test_quadratic_model.py:302]: I think this is better to have with an integer value `!=` 1. If the bug was not fixed, this test would not catch it.

Comment (alexzucca90) [reply tests/test_quadratic_model.py:302]: Actually, I'm wrong... there would be linear term = 0 and the energy would be 0, sorry

**Review by alexzucca90 — APPROVED** [review]

**Pull request merged**